homeassistant/components/bayesian/__init__.py

```diff
@@ -1,6 +1,51 @@
 """The bayesian component."""
 
-from homeassistant.const import Platform
+import logging
 
-DOMAIN = "bayesian"
-PLATFORMS = [Platform.BINARY_SENSOR]
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import discovery
+from homeassistant.helpers.typing import ConfigType
+
+from .binary_sensor import PLATFORM_SCHEMA  # noqa: F401
+from .const import DOMAIN, PLATFORMS
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
+    """Set up the Bayesian integration from YAML."""
+
+    for platform in PLATFORMS:
+        hass.async_create_task(
+            discovery.async_load_platform(hass, platform, DOMAIN, {}, config)
+        )
+    return True
```

Comment:
What's the reason for adding this? I don't this PR should introduce setup via yaml from the integration key.

Reply:
When I defined `async_setup_entry` YAML configs no longer loaded until I defined `async_setup`.

Reply:
This is loaded either by platform yaml or from config entry.
We should remove this as the usage of yaml in the platform should remain.

Reply:
If I remove `async_setup` when I have a YAML bayesian entry I get this error on restart
`2024-10-23 21:16:20.426 ERROR (MainThread) [homeassistant.setup] The 'bayesian' integration does not support YAML setup, please remove it from your configuration`

Reply:
That's because for some reason you have put `bayesian:` in your configuration.yaml.
It loads from a config entry or platform yaml (`binary_sensor: `) so that's something you did locally only.

Reply:
Nope, my entire config.yaml is here:
```

# Loads default set of integrations. Do not remove.
default_config:

# Load frontend themes from the themes folder
frontend:
  themes: !include_dir_merge_named themes

automation: !include automations.yaml
script: !include scripts.yaml
scene: !include scenes.yaml

logger:
  default: info
  logs:
    homeassistant.components.cloud: debug

# Example configuration.yaml entry
binary_sensor:
  name: "Paulus Home"
  platform: "bayesian"
  device_class: "presence"
  prior: 0.5
  probability_threshold: 0.9
  observations:
    - platform: template
      value_template: >
        {{is_state('input_boolean.ic','off') and ((as_timestamp(now()) - as_timestamp(states.input_boolean.ic.last_changed)) > 300)}}
      prob_given_true: 0.05
      prob_given_false: 0.99
```

Reply:
Some digging finds this logic
https://github.com/home-assistant/core/blob/69c6dd0cf2622c04c50d5bcaa55f4072121ba41c/homeassistant/setup.py#L226-L231

So it is the absence of a CONFIG_SCHEMA in `__init__.py` seems to be at least part of the issue

Reply:
Fixed in 08f277d
Importing the platform schema into init was confusing things

Reply:
Fixed in 08f277d